nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -169,18 +169,21 @@ let
       # Create a directory for exchanging data with the VM.
       mkdir -p "$TMPDIR/xchg"
 
-      ${lib.optionalString cfg.useBootLoader
+      ${lib.optionalString cfg.useEFIBoot
       ''
+        # Expose EFI variables, it's useful even when we are not using a bootloader (!).
+        # We might always be interested in a UEFI environment even if we directboot, hence
```

Comment:
It would be cool if you could add an example of why this is actually useful.

Reply:
UEFI Netboot tests! :D

Reply:
UEFI can:
- PXE boot
- boot through the option ROM of your NIC (iPXE)

Reply:
Yes right, completely missed that. Also just booting efi binaries that you somehow get via iPXE is a good use case for that, but then we're not really `direct booting` anymore, are we?

Reply:
> It would be cool if you could add an example of why this is actually useful.

This is a real stretch, but: you might want test a `kexec`-based bootloader (like `petitboot`) booting a final kernel which expects EFI variables.  So when building the VM, you want it to have an EFI partition even if the directbooted first kernel ignores it.